vector/src/main/java/im/vector/app/features/spaces/SpaceListFragment.kt

```diff
@@ -147,13 +150,22 @@ class SpaceListFragment :
     }
 
     override fun invalidate() = withState(viewModel) { state ->
-        when (state.asyncSpaces) {
+        when (val spaces = state.asyncSpaces) {
             Uninitialized,
             is Loading -> {
                 views.stateView.state = StateView.State.Loading
                 return@withState
             }
-            is Success -> views.stateView.state = StateView.State.Content
+            is Success -> {
+                views.stateView.state = StateView.State.Content
+                if (spaces.invoke().isEmpty()) {
+                    views.spacesEmptyGroup.isVisible = true
+                    views.groupListView.isVisible = false
+                } else {
+                    views.spacesEmptyGroup.isVisible = false
+                    views.groupListView.isVisible = true
+                }
+            }
```

Comment:
About StateView, the correct code would be:

```kotlin
            is Success -> {
                if (spaces.invoke().isEmpty()) {
                    views.stateView.state = StateView.State.Empty(...)
                } else {
                    views.stateView.state = StateView.State.Content
                }
            }
```


But maybe there are specific needs here? If this is the `spaces_empty_button`, maybe we could improve the StateView to integrate it. (can be done later, to not block this.)

Reply:
`StateView` has exact implementation for `Empty` state and I hadn't time to extend it to accept custom views (it rises some problems like callbacks for button). So in this case I had to place custom empty state near content view and switch between them. `StateView` her is mostly for loading state

Reply:
OK; thanks for explaining.